pycqed/analysis_v2/timedomain_analysis.py

```diff
@@ -291,6 +313,24 @@ def process_data(self):
         self.proc_data_dict['meas_results_per_qb'] = \
             meas_results_per_qb
 
+    def process_data(self):
+        super().process_data()
+        self.data_filter = self.get_param_value('data_filter')
+        self.data_with_reset = False
+        if self.data_filter is None:
+            if 'preparation_params' in self.metadata:
+                if 'active' in self.metadata['preparation_params'].get(
+                        'preparation_type', 'wait'):
+                    reset_reps = self.metadata['preparation_params'].get(
+                        'reset_reps', 1)
+                    data_filter = lambda x: x[reset_reps::reset_reps+1]
+                    self.data_with_reset = True
+        if self.data_filter is None:
+            self.data_filter = lambda x: x
+
+        self.create_sweep_points_dict()
+        self.create_meas_results_per_qb()
+
```

Comment:
New structure looks good! (Haven't checked every detail of it.)

Reply:
👍